arch/SPECS/wazuh-agent/arch/wazuh.install

```diff
@@ -80,3 +83,29 @@ post_install() {
 
     common
 }
+
+pre_remove() {
+    systemctl disable wazuh-agent.service 2>/dev/null || true
+    systemctl stop wazuh-agent.service 2>/dev/null || true
```

Comment:
Could you use the wazuh-control binary to stop the service? An example would be like this https://github.com/wazuh/wazuh-packages/blob/f86b8425cb2327f219e1f83d5d363d149b2a52b4/debs/SPECS/wazuh-agent/debian/prerm#L23 where we check with systemctl, service and finally with wazuh-control

I just generated new packages with the branch of your repository and now I see that the stop of the daemons fails again, this is essential for the upgrade since it must stop them before doing it.

<details><summary>Remove daemons error</summary>

```
[root@archlinux vagrant]# pacman -U wazuh-agent-4.5.0-1-x86_64.pkg.tar.zst 
loading packages...
resolving dependencies...
looking for conflicting packages...

Packages (1) wazuh-agent-4.5.0-1

Total Installed Size:  26.10 MiB

:: Proceed with installation? [Y/n] Y
(1/1) checking keys in keyring                                 [##################################] 100%
(1/1) checking package integrity                               [##################################] 100%
(1/1) loading package files                                    [##################################] 100%
(1/1) checking for file conflicts                              [##################################] 100%
(1/1) checking available disk space                            [##################################] 100%
:: Processing package changes...
(1/1) installing wazuh-agent                                   [##################################] 100%
:: Running post-transaction hooks...
(1/3) Creating system user accounts...
(2/3) Reloading system manager configuration...
(3/3) Arming ConditionNeedsUpdate...
[root@archlinux vagrant]# nano /var/ossec/etc/ossec.conf 
[root@archlinux vagrant]# ps aux | grep wazuh
root        4148  0.0  0.2   4092  2432 pts/0    S+   14:23   0:00 grep wazuh
[root@archlinux vagrant]# /var/ossec/bin/wazuh-control start
Starting Wazuh v4.5.0...
Started wazuh-execd...
Started wazuh-agentd...
Started wazuh-syscheckd...
Started wazuh-logcollector...
Started wazuh-modulesd...
Completed.
[root@archlinux vagrant]# ps aux | grep wazuh
root        4177  0.0  0.2  24932  2744 ?        Sl   14:23   0:00 /var/ossec/bin/wazuh-execd
wazuh       4188  0.0  0.6 172448  6216 ?        Sl   14:23   0:00 /var/ossec/bin/wazuh-agentd
root        4199  0.0  0.5  99108  5604 ?        SNl  14:23   0:00 /var/ossec/bin/wazuh-syscheckd
root        4208  0.0  1.2 336064 12340 ?        Sl   14:23   0:00 /var/ossec/bin/wazuh-logcollector
root        4224  0.5  1.3 521200 13212 ?        Sl   14:23   0:00 /var/ossec/bin/wazuh-modulesd
root        4263  0.0  0.2   4092  2304 pts/0    S+   14:23   0:00 grep wazuh
[root@archlinux vagrant]# pacman -R wazuh-agent
checking dependencies...

Packages (1) wazuh-agent-4.5.0-1

Total Removed Size:  26.10 MiB

:: Do you want to remove these packages? [Y/n] Y
:: Processing package changes...
(1/1) removing wazuh-agent                                     [##################################] 100%
warning: /var/ossec/etc/ossec.conf saved as /var/ossec/etc/ossec.conf.pacsave
:: Running post-transaction hooks...
(1/2) Reloading system manager configuration...
(2/2) Arming ConditionNeedsUpdate...
[root@archlinux vagrant]# ps aux | grep wazuh
root        4177  0.0  0.2  24932  2744 ?        Sl   14:23   0:00 /var/ossec/bin/wazuh-execd
wazuh       4188  0.0  0.6 172448  6216 ?        Sl   14:23   0:00 /var/ossec/bin/wazuh-agentd
root        4199  0.0  0.5  99108  5604 ?        SNl  14:23   0:00 /var/ossec/bin/wazuh-syscheckd
root        4208  0.0  1.2 336064 12340 ?        Sl   14:23   0:00 /var/ossec/bin/wazuh-logcollector
root        4224  0.1  1.3 521200 13212 ?        Sl   14:23   0:00 /var/ossec/bin/wazuh-modulesd
root        4311  0.0  0.2   4092  2452 pts/0    S+   14:23   0:00 grep wazuh
[root@archlinux vagrant]#
```

</details>

Also i found that the deployment variables are not working and the package install show errors:

- https://documentation.wazuh.com/current/user-manual/deployment-variables/deployment-variables-linux.html

<details><summary>Deployment variables error</summary>

```
[root@archlinux vagrant]# WAZUH_MANAGER="192.168.57.102" pacman -U wazuh-agent-4.5.0-1-x86_64.pkg.tar.zst 
loading packages...
resolving dependencies...
looking for conflicting packages...

Packages (1) wazuh-agent-4.5.0-1

Total Installed Size:  26.10 MiB

:: Proceed with installation? [Y/n] Y
(1/1) checking keys in keyring                                 [##################################] 100%
(1/1) checking package integrity                               [##################################] 100%
(1/1) loading package files                                    [##################################] 100%
(1/1) checking for file conflicts                              [##################################] 100%
(1/1) checking available disk space                            [##################################] 100%
:: Processing package changes...
(1/1) installing wazuh-agent                                   [##################################] 100%
touch: cannot touch '/logs/ossec.log': No such file or directory
chmod: cannot access '/logs/ossec.log': No such file or directory
chown: cannot access '/logs/ossec.log': No such file or directory
grep: /etc/ossec.conf: No such file or directory
grep: /etc/ossec.conf: No such file or directory
sed: can't read /etc/ossec.conf: No such file or directory
/var/ossec/packages_files/agent_installation_scripts/src/init/register_configure_agent.sh: line 58: /logs/ossec.log: No such file or directory
:: Running post-transaction hooks...
(1/3) Creating system user accounts...
(2/3) Reloading system manager configuration...
(3/3) Arming ConditionNeedsUpdate...
[root@archlinux vagrant]# grep "MANAGER_IP" /var/ossec/etc/ossec.conf
      <address>MANAGER_IP</address>
[root@archlinux vagrant]#
```

</details>

Some examples on how to manage these variables can be found in the SPECS of deb and rpm packages generation process

https://github.com/wazuh/wazuh-packages/blob/f86b8425cb2327f219e1f83d5d363d149b2a52b4/debs/SPECS/wazuh-agent/debian/postinst#L125
https://github.com/wazuh/wazuh-packages/blob/f86b8425cb2327f219e1f83d5d363d149b2a52b4/rpms/SPECS/wazuh-agent.spec#L253